### PR TITLE
Add helper function to validate a webhook that is originated from Eas…

### PR DESCRIPTION
**Pull request by jchen293**
# Description

Adds a new function to validate if a webhook originated from EasyPost with the stored Webhook secret, if they do the function will return `event_body`, otherwise an error will be thrown.

# Testing

Add three unit tests

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: lib/EasyPost/Webhook.php [hunk 1]
```diff
@@ -82,4 +84,34 @@ public static function create($params = null, $apiKey = null)
...
+    /**
+     * Validate a webhook originated from EasyPost by comparing the HMAC header to a shared secret.
+     * This function will return True if the webhook originated from EasyPost and False if it did not.
+     *
+     * @param mixed $webhook_body
+     * @param mixed $headers
+     * @param string $webhook_secret
+     * @return boolean
+     */
+    public static function validateWebhook($webhook_body, $headers, $webhook_secret)
```
Comment (Justintime50) [inline lib/EasyPost/Webhook.php:97]: Function params also need to be camel cased (the rest will get updated in the rewrite)

```suggestion
    public static function validateWebhook($webhookBody, $headers, $webhookSecret)
```

Comment (jchen293) [reply lib/EasyPost/Webhook.php:97]: It's still in progress, I will correct this

File: lib/EasyPost/Webhook.php [hunk 2]
```diff
@@ -82,4 +84,37 @@ public static function create($params = null, $apiKey = null)
...
+     * @param string $webhookSecret
+     * @return mixed
+     */
+    public static function validateWebhook($eventBody, $headers, $webhookSecret)
+    {
+        $easypostHmacSignature = $headers["X-Hmac-Signature"] ?? null;
+
+        if ($easypostHmacSignature != null) {
+            $normalizedSecret = Normalizer::normalize($webhookSecret, Normalizer::FORM_KD);
+            $encodedSecret = mb_convert_encoding(utf8_encode($normalizedSecret), "ISO-8859-1", "UTF-8");
```
Comment (Justintime50) [inline lib/EasyPost/Webhook.php:104]: This still isn't perfectly ideal as there could be slight alterations depending on the input. Let's fix it to be uniform with the other libraries.
```suggestion
            $encodedSecret = mb_convert_encoding($normalizedSecret, 'UTF-8');
```

**Review by Justintime50 — CHANGES_REQUESTED** [review]

File: test/EasyPost/WebhookTest.php [hunk 3]
```diff
@@ -131,4 +132,61 @@ public function testDelete()
         // This endpoint/method does not return anything, just make sure the request doesn't fail
         $this->assertInstanceOf('\EasyPost\Webhook', $response);
     }
+
+    /**
+     * Test a webhook signature that is originated from EasyPost by comparing the HMAC header
+     * to a shared secret.
+     *
+     * @return void
```
[the remark this answers is not in the document]
Comment (jchen293) [reply test/EasyPost/WebhookTest.php:140]: FYI, all of the unit tests have a return type in the docstring.

[the remark this answers is not in the document]
Comment (Justintime50) [reply test/EasyPost/WebhookTest.php:140]: Indeed they do. Can you add an internal task to remove these? They shouldn't be needed for a test context.

File: lib/EasyPost/Webhook.php (hunk 2 above)
Comment (jchen293) [reply lib/EasyPost/Webhook.php]: fixed

**Review by Justintime50 — APPROVED** [review]

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**